testsuite/experimental/src/test/java/io/smallrye/health/deployment/DependentCallRecorder.java

```diff
@@ -0,0 +1,8 @@
+package io.smallrye.health.deployment;
+
+public class DependentCallRecorder {
+
+    public static boolean healthCheckPreDestroyCalled = false;
```

Comment:
These fields should be `volatile`, just to be sure... but that's just nitpicking ;-).